File: src/main/java/com/server/computerscience/chatbot/dto/request/ChatGptRestRequestDto.java
package com.server.computerscience.chatbot.dto.request;

import java.util.List;
import java.util.stream.Collectors;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.ToString;

@Getter
@ToString
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ChatGptRestRequestDto {
	private String model;
	private List<ChatGptMessageDto> messages;

	public static ChatGptRestRequestDto from(String model, List<String> prompts) {
		return ChatGptRestRequestDto.builder()
			.model(model)
			.messages(prompts.stream()
				.map(ChatGptMessageDto::fromUserText)
				.collect(Collectors.toList()))
			.build();
	}
}

File: src/main/java/com/server/computerscience/chatbot/dto/request/ChatGptRestRequestDto.kt
package com.server.computerscience.chatbot.dto.request

data class ChatGptRestRequestDto(
    val model: String? = null,
    val messages: List<ChatMessageDto?>? = null
) {
    companion object {
        @JvmStatic
        fun from(model: String?, chatMessages: List<ChatMessageDto?>?): ChatGptRestRequestDto {
            return ChatGptRestRequestDto(
                model = model,
                messages = chatMessages
            )
        }
    }
}
